app/models/password_recovery_form.rb

```diff
@@ -0,0 +1,22 @@
+class PasswordRecoveryForm
+  include ActiveModel::Model
+
+  attr_reader :code, :password, :password_confirmation
+  attr_accessor :email
+
+  validates_presence_of :code, :email, :password, :password_confirmation
```

Comment:
do we know the format of `:code`, min/max character? we could add the validation here too